doc/api-management.md

```diff
@@ -0,0 +1,145 @@
+# API Management
+
+> Guidance below for Theia framework. End products will get better development experience by using private visibility.
+
+- [**Stability**](#stability)
+  - [**Experimental**](#experimental)
+  - [**Stable**](#stable)
+- [**Finalization**](#finalization)
+- [**Deprecation**](#deprecation)
+
+Theia is a framework embracing openness, extensibility, and customizability as much as possible:
+- API defaults to public for clients.
+- API defaults to protected for extenders.
+- Language constructions prohibiting runtime access to internals are never used and so on.
+
+Usually, the version management is built around API visibility.
+Particularly, if a public API is broken then the major release is required.
```

Comment:
`is broken then the major release is required.` -> `is broken then, a new major release is required.`